scrabble.js

```diff
@@ -1,8 +1,173 @@
-var Scrabble = function() {};
+var Scrabble = function() {
+    // due to how the specs are written, this score table does not account for the 50 point bonus for 7-letter words...
+    this.scoreTable = [
+        [1, "A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+        [2, "D", "G"],
+        [3, "B", "C", "M", "P"],
+        [4, "F", "H", "V", "W", "Y"],
+        [5, "K"],
+        [8, "J", "X"],
+        [10, "Q", "Z"]
+    ];
+};
+
+// score(word): returns the total score value for the given word. The word is input as a string (case insensitive).
+Scrabble.prototype.score = function(word) {
+    var wordScore = 0;
+    var nonAlphaChar = 0;
+    var alphaChars = ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T", "D", "G", "B", "C", "M", "P", "F", "H", "V", "W", "Y", "K", "J", "X", "Q", "Z"];
+
+    var scoreTable = this.scoreTable; //this local variable is necessary to bypass the error that otherwise would be thrown in line 18 because the scope of 'this' changes inside anonymous functions.
+
+    if (word.length > 7) {
+        return 0;
+    }
+
+    var wordAsArray = word.toUpperCase().split("");
+    wordAsArray.forEach(function(letter) {
+        if (alphaChars.includes(letter)) {
+            scoreTable.forEach(function(scoreRow) {
+                if (scoreRow.includes(letter)) {
+                    wordScore += scoreRow[0];
+                }
+            });
+        } else if (alphaChars.includes(letter) === false) {
+            nonAlphaChar += 1; // this is my second or fourth idea how to handle words with non-alpha characters, since the forEach function behaves differently than I originally expected.
+        }
+    });
+    if (nonAlphaChar !== 0) {
+        return 0;
+    } else {
+        return wordScore;
+    }
+};
+
+// highestScoreFrom(arrayOfWords): returns the word in the array with the highest score (based on specs).
+Scrabble.prototype.highestScoreFrom = function(arrayOfWords) {
+    var scores = [];
+    for (var i = 0; i < arrayOfWords.length; i++) {
+        aScore = this.score(arrayOfWords[i]);
+        scores.push(aScore);
+    }
+
+    // find the highest score
+    var greatestVal = 0;
+    for (var j = 0; j < scores.length; j++) {
+        if (scores[j] > greatestVal) {
+            greatestVal = scores[j];
+        }
+    }
+
+    // to handle "tie" situations, creating a new array with the words whose score is the greatestVal
+    var highestScoreWords = [];
+    for (var k = 0; k < arrayOfWords.length; k++) {
+        if (this.score(arrayOfWords[k]) === greatestVal) {
```

Comment:
It would be more efficient to use the `scores` array here instead of re-calculating the word scores